### PR TITLE
googletest

### PR DESCRIPTION
**Pull request by scopatz**
This is an initial google test recipe. A couple annoying things about this project is that it requires Python 2 for its recipe and doesn't have an install target.

### CODE REVIEW

**Comment by scopatz** [comment]
Woohoo it builds!

**Comment by jakirkham** [comment]
So, I think I had tried to do this once, but convinced myself it was a bad idea. Part of the reason was I read this in `googletest`'s [`Makefile.am`]( https://github.com/google/googletest/blob/release-1.7.0/Makefile.am#L296-L306 ). Also, there is this point in there [FAQ]( https://github.com/google/googletest/blob/d225acc90bc3a8c420a9bcd1f033033c1ccd7fe0/googletest/docs/V1_7_FAQ.md#why-is-it-not-recommended-to-install-a-pre-compiled-copy-of-google-test-for-example-into-usrlocal ). So, basically, I just download this as part of the build process as needed. Here is an [example]( https://github.com/conda-forge/protobuf-feedstock/blob/1902c3d1ee3bb505dc3a8501ad93a74d43c76e14/recipe/build.sh#L19-L21 ) with `protobuf`. I hope I have saved you some suffering here.

**Comment by jakirkham** [comment]
Ouch, sorry, not soon enough I guess. 😦

**Comment by scopatz** [comment]
@jakirkham - yeah, I know what they recommend and why. Still, I think that there are a couple of reasons to have this as a package. 

1. Other package managers have it, [libgtest-dev](https://packages.debian.org/sid/libgtest-dev), [portage](https://packages.gentoo.org/packages/dev-cpp/gtest), etc. Though this goes both ways since not all other package managers have it, 
2. Since we are in a controlled environment on conda-forge the One-Definition rule is less likely to be an issue when creating other recipes that are dependent on this package, and
3. Packages that vendor googletest for their own test suite are going to include it anyway and they will ignore this package, even if it exists.  

So I don't see the harm in providing this as on option to packagers and developers.  If it works, great! If it doesn't they may have had to distribute it anyway...

**Comment by jakirkham** [comment]
Sure. I think that is reasonable. I'm ok with giving this a shot. 👍

Before we go ahead though, could we add a warning somewhere explaining that if it doesn't work one might need to build it from scratch in their build process with a link to instructions (probably from the googletest repo) on how to do this.

File: recipes/googletest/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    - test -f "${PREFIX}/lib/libgtest_main{{ libext }}"
+
+about:
+  home: https://github.com/google/googletest
+  license: BSD 3-clause
+  summary: "Google C++ Testing Framework"
+
+extra:
+  recipe-maintainers:
+    - scopatz
```
Comment (jakirkham) [inline recipes/googletest/meta.yaml]: Please add me.

File: recipes/googletest/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  version: {{ version }}
+
+source:
+  fn: release-{{ version }}.tar.gz
+  url: https://github.com/google/googletest/archive/release-{{ version }}.tar.gz
+  sha256: f73a6546fdf9fce9ff93a5015e0333a8af3062a152a9ad6bcb772c96687016cc
+
+build:
+  number: 0
+  skip: true  # [win]
```
Comment (jakirkham) [inline recipes/googletest/meta.yaml]: Change to `skip: true  # [win or py3k]`.

File: recipes/googletest/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+requirements:
+  build:
+    - python <3  # tests are Python 2k only :(
```
Comment (jakirkham) [inline recipes/googletest/meta.yaml]: The `skip` change above will make this unnecessary. Using `skip` is the preferred way for handling platform, Python, or NumPy constraints. Other constraints are fine to handle this way.

**Comment by jakirkham** [comment]
Added a few other comments that we should address, but they should be pretty easy.

Should we do `googlemock` as well?

**Comment by ocefpaf** [comment]
Ping  @scopatz this seems like a low hanging fruit to merge. Can you take a quick look at the comments? If you don't have the time I can take this over and make the modifications.

**Comment by jakirkham** [comment]
Friendly nudge @scopatz. Would be happy to merge this when the very minor errata above is addressed? Will you have time to do so or would you rather I take over and finish it?

File: recipes/googletest/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  fn: release-{{ version }}.tar.gz
+  url: https://github.com/google/googletest/archive/release-{{ version }}.tar.gz
+  sha256: f73a6546fdf9fce9ff93a5015e0333a8af3062a152a9ad6bcb772c96687016cc
+
+build:
+  number: 0
+  skip: true  # [win]
+
+requirements:
+  build:
```
Comment (jakirkham) [inline recipes/googletest/meta.yaml]: Please add `toolchain`.

**Comment by scopatz** [comment]
Sorry about the delay here.  I have made the updates.

**Comment by jakirkham** [comment]
I think the CI failures here were spurious (download issues with Anaconda.org). So have restarted them.

File: recipes/googletest/meta.yaml (hunk 1 above)
Comment (jakirkham) [inline recipes/googletest/meta.yaml]: Done.

File: recipes/googletest/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  sha256: f73a6546fdf9fce9ff93a5015e0333a8af3062a152a9ad6bcb772c96687016cc
+
+build:
+  number: 0
+  skip: true  # [win or py3k]
+
+requirements:
+  build:
+    - cmake
+    - toolchain
```
Comment (jakirkham) [inline recipes/googletest/meta.yaml]: We had `python` before, but we seem to have lost it. Should we be adding it back?

File: recipes/googletest/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+test:
+  commands:
+    - test -f "${PREFIX}/lib/libgtest{{ libext }}"
+    - test -f "${PREFIX}/lib/libgtest_main{{ libext }}"
+
+about:
+  home: https://github.com/google/googletest
+  license: BSD 3-clause
```
Comment (jakirkham) [inline recipes/googletest/meta.yaml]: Please add `license_file: LICENSE`.

**Comment by scopatz** [comment]
Updated!

**Comment by jakirkham** [comment]
cc @SylvainCorlay

File: recipes/googletest/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+  license_file: googletest/LICENSE
```
Comment (jakirkham) [inline recipes/googletest/meta.yaml]: This is for the `1.8.0` release. It is just `LICENSE` for `1.7.0`.

**Pull request closed without merging**